v1/plugins/logs/mask_test.go

```diff
@@ -487,22 +485,61 @@ func TestMaskRuleMask(t *testing.T) {
 			exp:   `{"input": {"foo": [{"baz": 1}]}}`,
 		},
 		{
-			note: "erase: undefined array: remove element",
+			note: "erase: array: remove element",
 			ptr: &maskRule{
 				OP:   maskOPRemove,
 				Path: "/input/foo/0",
 			},
 			event: `{"input": {"foo": [1]}}`,
-			exp:   `{"input": {"foo": [1]}}`,
+			exp:   `{"input": {"foo": []}, "erased": ["/input/foo/0"]}`,
 		},
 		{
-			note: "upsert: unsupported nested object type (array) #2",
+			note: "erase: array: remove element with deeper nesting",
 			ptr: &maskRule{
-				OP:   maskOPUpsert,
-				Path: "/input/foo/0",
+				OP:   maskOPRemove,
+				Path: "/input/foo/0/bar/1",
+			},
+			event: `{"input": {"foo": [{"bar": [1, 2]}]}}`,
+			exp:   `{"input": {"foo": [{"bar": [1]}]}, "erased": ["/input/foo/0/bar/1"]}`,
+		},
+		{
+			note: "erase: array: remove element that does not exist",
```

Comment:
What happens if you upsert at a path like this? Presumably, it'll fail rather than create a sparse array? But might be good to test for.

Reply:
Thanks, added in [5e6bb05](https://github.com/open-policy-agent/opa/pull/7246/commits/5e6bb0558474a35bd07a173560d06a32e77b7154)